Fixes #31342 - add Category and template step to job wizard

MariaAga opened the pull request:
Depends on https://github.com/theforeman/foreman_remote_execution/pull/546
There is a typeahead issue in PF4 that is fixed in 2020.14, so user can't search for a template in the select
![Screenshot from 2020-11-19 16-34-24](https://user-images.githubusercontent.com/30431079/99679874-1d912300-2a85-11eb-9f77-913a90a11d0d.png)

adamruzicka commented on webpack/react_app/components/JobWizard/steps/CategoryAndTemplate/CategoryAndTemplate.js:
```diff
@@ -0,0 +1,77 @@
+    setJobTemplate(null);
+  };
+  return (
+    <>
```
An empty tag?

MariaAga replied:
https://reactjs.org/docs/fragments.html#short-syntax
Reacts asks that the `return` only returns one element so I'm wrapping all the elements

adamruzicka replied:
Oh, wow. I was aware of `React.Fragment`, but not the short syntax. Nevermind then

adamruzicka commented:
The input fields tend to get a bit wide, maybe some kind of limit would be in order

![image](https://user-images.githubusercontent.com/7326770/99955080-06ec1400-2d84-11eb-86e3-347160e74544.png)

MariaAga commented:
Made the snapshots smaller and added a width prop.
There is a bug in PF4 that was fixed in newer versions that caused the width to be ignored, so once `foreman-js` gets a new version I'll update it
Preview of the new fixed size:
![Screenshot from 2020-11-25 17-53-15](https://user-images.githubusercontent.com/30431079/100252498-9b06d880-2f48-11eb-93ce-8141fc997a50.png)

MariaAga commented:
After working more on the wizard I decided it was better to add a max-width for the wizard content instead limiting each fields width.

MariaAga commented on webpack/JobWizard/steps/CategoryAndTemplate/CategoryAndTemplate.js:
```diff
@@ -0,0 +1,77 @@
+          setValue={onSelectCategory}
+          value={selectedCategory}
+        />
+        <GroupedSelectField
```
It's a button for clearing the search text and I don't think I can get rid of it, I can remove the option to search

adamruzicka replied:
Oh well, nevermind then

MariaAga commented on webpack/JobWizard/steps/CategoryAndTemplate/CategoryAndTemplate.js:
```diff
@@ -0,0 +1,77 @@
+
+  const onSelectCategory = newCategory => {
+    setCategory(newCategory);
+    setJobTemplate(null);
```
Sure

MariaAga commented:
Rebased, updated vendor to get a newer pf4 version, the new version doesn't show an console error when selecting a category.

adamruzicka commented:
Looks good. @laviro / @amirfefer could any of you take a look at the code please?

MariaAga commented on webpack/JobWizard/steps/CategoryAndTemplate/CategoryAndTemplate.js:
```diff
@@ -0,0 +1,77 @@
+  return (
+    <>
+      <Title headingLevel="h2">{__('Category And Template')}</Title>
+      <p className="wizard-sub-title">{__('All fields are required.')}</p>
```
Where?

laviro replied:
https://codesandbox.io/s/3c3cl?file=/index.js:328-366

MariaAga replied:
Yeah, thanks for the example!

MariaAga commented on webpack/JobWizard/steps/CategoryAndTemplate/index.js:
```diff
@@ -0,0 +1,82 @@
+          }),
+          handleSuccess: response =>
+            setJobTemplate(
+              Number(filterJobTemplates(response?.data?.results)[0]?.id) || null
```
I want it to be clear the a template is not selected with using null instead of a number

MariaAga commented on webpack/JobWizard/steps/CategoryAndTemplate/index.js:
```diff
@@ -0,0 +1,82 @@
+    }
+  }, [category, dispatch, setJobTemplate]);
+
+  const jobTemplates = useSelector(selectJobTemplates);
```
I want to have CategoryAndTemplate be a "clean" component without any redux

MariaAga commented:
Thanks @laviro! I addressed the comments, and added a small scss fix to prevent an unnecessary scroll bar from showing 
before:
![Screenshot from 2021-03-15 13-31-20](https://user-images.githubusercontent.com/30431079/111147066-c7e00980-8592-11eb-84c0-b0eb3ca088cf.png)
after:
![Screenshot from 2021-03-15 13-31-29](https://user-images.githubusercontent.com/30431079/111147059-c6aedc80-8592-11eb-8076-fd761dc686b3.png)

laviro reviewed (approved):
Thanks @MariaAga! Looks good to me :)
Failures of ruby tests do not seem to be related

adamruzicka commented:
Could you please rebase? It should make the test failures go away

MariaAga commented:
rebased

MariaAga commented:
Hi @adamruzicka, can this be merged? :)

adamruzicka commented:
Certainly can, I was just looking at this before the scrum :)

adamruzicka commented:
Thank you @MariaAga & @laviro !
